src/funcx_common/messagepack/message_types/result.py

```diff
@@ -0,0 +1,51 @@
+from __future__ import annotations
+
+import enum
+import typing as t
+import uuid
+
+import pydantic
+
+from .base import Message, meta
+
+
+class ResultErrorCode(enum.Enum):
+    ContainerError = "ContainerError"
```

Comment:
@ryanchard Please check if this makes sense. We don't report such an error now, but we should be reporting an error when a task fails due to issues with the requested container.

Reply:
Also, as regards this enum:
- If there are other known categories of error we want to capture, the sooner we list them here the better
- If there's some known mapping where certain errors indicate that `is_system_error=t/f`, then we can either (1) validate that they match with a validator or (2) turn `is_system_error` into a property computed from the `code`